projects/packages/assets/src/class-initial-state.php

```diff
@@ -0,0 +1,192 @@
+
+		$initial_state = wp_json_encode(
+			$initial_state,
+			JSON_UNESCAPED_SLASHES | JSON_HEX_TAG | JSON_HEX_AMP | JSON_UNESCAPED_UNICODE
+		);
+
+		wp_add_inline_script(
+			self::SCRIPT_HANDLE,
+			sprintf( 'var JETPACK_INITIAL_STATE = %s;', $initial_state ),
+			'before'
+		);
+	}
+
+	/**
+	 * Get the admin initial state.
+	 *
+	 * @return array
+	 */
+	protected static function get_admin_initial_state() {
+
+		global $wp_version;
+
+		$state = array(
+			'site' => array(
+				'admin_url'    => esc_url_raw( admin_url() ),
+				'date_format'  => get_option( 'date_format' ),
+				'icon'         => self::get_site_icon(),
+				'is_multisite' => is_multisite(),
+				'plan'         => array(
+					// The properties here should be updated by the consumer package/plugin.
+					// It includes properties like 'product_slug', 'features', etc.
+					'product_slug' => '',
+				),
+				'rest_nonce'   => wp_create_nonce( 'wp_rest' ),
+				'rest_root'    => esc_url_raw( rest_url() ),
+				'title'        => self::get_site_title(),
+				'wp_version'   => $wp_version,
+				'wpcom'        => array(
+					// This should contain the connected site details like blog_id, is_atomic etc.
+					'blog_id' => 0,
+				),
+			),
+			'user' => array(
+				'current_user' => self::get_current_user_data(),
+			),
+		);
+
+		/**
+		 * Filter the admin initial state.
+		 *
+		 * When using this filter, ensure that the data is added only if it is used by some script.
+		 * This filter may be called on almost every admin page load. So, one should check if the data is needed/used on that page.
+		 * For example, the social (publicize) data is used only on Social admin page, Jetpack settings page and the post editor.
+		 * So, the social data should be added only on those pages.
+		 *
+		 * @param array $state The initial state.
+		 */
+		return apply_filters( 'jetpack_admin_js_initial_state', $state );
+	}
+
+	/**
+	 * Get the admin initial state.
+	 *
+	 * @return array
+	 */
+	protected static function get_public_initial_state() {
+
+		$state = array(
+			'site' => array(
+				'icon'  => self::get_site_icon(),
+				'title' => self::get_site_title(),
+			),
+		);
+
+		/**
+		 * Filter the public initial state.
+		 *
+		 * See the docs for `jetpack_admin_js_initial_state` filter for more information.
+		 *
+		 * @param array $state The initial state.
+		 */
+		return apply_filters( 'jetpack_public_js_initial_state', $state );
+	}
+
+	/**
+	 * Get the site title.
+	 *
+	 * @return string
+	 */
+	protected static function get_site_title() {
+		$title = get_bloginfo( 'name' );
+
+		return $title ? $title : esc_url_raw( ( get_site_url() ) );
+	}
+
+	/**
+	 * Get the site icon.
+	 *
+	 * @return string
+	 */
+	protected static function get_site_icon() {
+		if ( ! has_site_icon() ) {
+			return '';
+		}
+
+		return apply_filters( 'jetpack_photon_url', get_site_icon_url(), array( 'w' => 64 ) );
```

Comment:
Let's add a filter docblock here as well please.

Reply:
Updated. Thank you.